### PR TITLE
[BugFix] rename element_memory_usage to reference_memory_usage and fix incorrect result

### PR DESCRIPTION
**Pull request by silverbullet233**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

1. the name of `Column::element_memory_usage` is confusing, rename it to `Column::reference_memory_usage` and fix the error result introduced by #17184
2. after fix the first point, local_exchange should use bytes_usage instead of  element_memory_usage to calculate memory usage.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by silverbullet233** [comment]
run starrocks_admit_test

**Review by stdpain — DISMISSED** [review]

**Comment by silverbullet233** [comment]
run starrocks_admit_test

**Comment by silverbullet233** [comment]
run starrocks_admit_test

File: be/test/column/map_column_test.cpp [hunk 1]
```diff
@@ -1202,17 +1202,6 @@ PARALLEL_TEST(MapColumnTest, test_element_memory_usage) {
     column->append_datum(DatumMap{{1, 2}});
     column->append_datum(DatumMap{{3, 4}, {5, 6}});
 
-    ASSERT_EQ(42, column->Column::element_memory_usage());
```
Comment (fzhedu) [inline be/test/column/map_column_test.cpp:1205]: change `element_memory_usage` to `bytes_usage` to check `bytes_usage`?

File: be/test/column/binary_column_test.cpp [hunk 2]
```diff
@@ -655,25 +655,14 @@ PARALLEL_TEST(BinaryColumnTest, test_replicate) {
...
-PARALLEL_TEST(BinaryColumnTest, test_element_memory_usage) {
+PARALLEL_TEST(BinaryColumnTest, test_reference_memory_usage) {
     auto column = BinaryColumn::create();
     column->append("");
     column->append("1");
     column->append("23");
     column->append("456");
 
-    ASSERT_EQ(22, column->Column::element_memory_usage());
-
```
Comment (fzhedu) [inline be/test/column/binary_column_test.cpp:666]: change to `bytes_usage`?

**Review by satanson — DISMISSED** [review]

**Review by stdpain — DISMISSED** [review]

**Comment by stdpain** [comment]
run starrocks_admit_test

**Comment by stdpain** [comment]
run starrocks_be_unittest

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: be/test/column/binary_column_test.cpp (hunk 2 above)
Comment (silverbullet233) [reply be/test/column/binary_column_test.cpp:666]: > change to `bytes_usage`?

test_byte_size method is already existed, I think we don't need to check it in this method.

**Comment by silverbullet233** [comment]
run starrocks_admit_test

**Comment by silverbullet233** [comment]
run starrocks_be_unittest

**Comment by silverbullet233** [comment]
run starrocks_be_unittest

**Review by stdpain — APPROVED** [review]

**Review by satanson — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by trueeyu** [comment]
https://github.com/Mergifyio backport branch-3.0 branch-2.5

**Pull request merged**